### PR TITLE
Function to relabel nodes and edges

### PR DESCRIPTION
**Pull request by nwlandry**
This implements a function very similar to `networkx.convert_node_labels_to_integers`. but for both node and edge IDs.

### CODE REVIEW

File: tests/utils/test_utils.py [hunk 1]
```diff
@@ -25,3 +25,26 @@ def test_get_dual(dict5):
     assert dual[6] == [2, 3]
     assert dual[7] == [3]
     assert dual[8] == [3]
+
+
+def test_convert_labels_to_integers(hypergraph1):
```
Comment (leotrs) [inline tests/utils/test_utils.py:30]: Can you please add another test where the original hypergraph has nodes/edges labeled by non-consecutive integers? And/Or, a test where the original hypergraph has a single node labeled `0` but all other nodes are labeled by strings.

File: xgi/utils/utilities.py [hunk 2]
```diff
@@ -73,3 +74,51 @@ def load_xgi_data(dataset, nodetype=None, edgetype=None):
...
+        temp_H._node[id] = temp_H._node.pop(node)
+        temp_H._node_attr[id] = temp_H._node_attr.pop(node)
+        temp_H._node_attr[id].update({"label": node})
+        node_mapping[node] = id
+
+    for edge in list(temp_H.edges):
+        id = next(edge_id)
+        temp_H._edge[id] = temp_H._edge.pop(edge)
+        temp_H._edge_attr[id] = temp_H._edge_attr.pop(edge)
+        temp_H._edge_attr[id].update({"label": edge})
```
Comment (leotrs) [inline xgi/utils/utilities.py:115]: Why this instead of `temp_H._edge_attr[id]["label"] = edge`?

File: xgi/utils/utilities.py [hunk 3]
```diff
@@ -73,3 +74,51 @@ def load_xgi_data(dataset, nodetype=None, edgetype=None):
...
+    the old "label" values (if they exist) will be overwritten.
+    """
+    node_id = count()
+    edge_id = count()
+    node_mapping = dict()
+    edge_mapping = dict()
+    temp_H = H.copy()
+    for node in list(temp_H.nodes):
+        id = next(node_id)
+        temp_H._node[id] = temp_H._node.pop(node)
```
Comment (leotrs) [inline xgi/utils/utilities.py:106]: I'm pretty sure this is buggy. For example, if the original hypergraph has a node with id `0` (but other nodes have ids that are, say, strings), the first time this line is executed, the edges incident to `0` will be overwritten and can never be recovered.

My suggestion is to first build the `node_mapping` and `edge_mapping` dictionaries WITHOUT modifying the network, and then using those mappings to do the relabeling in a single loop. In this loop, you are trying to build the `node_mapping` at the same time as modifying the network, and that's what's causing the problem.

File: tests/utils/test_utils.py (hunk 1 above)
Comment (leotrs) [reply tests/utils/test_utils.py:30]: This will catch the bug I point out later.

**Review by leotrs — CHANGES_REQUESTED** [review]

File: xgi/utils/utilities.py (hunk 2 above)
Comment (nwlandry) [reply xgi/utils/utilities.py:115]: Good call.

File: xgi/utils/utilities.py (hunk 3 above)
Comment (nwlandry) [reply xgi/utils/utilities.py:106]: This is a good suggestion. Done.

File: xgi/utils/utilities.py [hunk 4]
```diff
@@ -73,3 +74,51 @@ def load_xgi_data(dataset, nodetype=None, edgetype=None):
     r = requests.get(index[dataset]["url"])
 
     return convert.dict_to_hypergraph(r.json(), nodetype=nodetype, edgetype=edgetype)
+
+
+def convert_labels_to_integers(H):
```
[the remark this answers is not in the document]
Comment (nwlandry) [reply xgi/utils/utilities.py:79]: Done.

File: tests/utils/test_utils.py (hunk 1 above)
Comment (nwlandry) [reply tests/utils/test_utils.py:30]: Done.

File: tests/conftest.py [hunk 5]
```diff
@@ -169,3 +169,11 @@ def hypergraph1():
     H.add_nodes_from(["a", "b", "c"])
     H.add_edges_from({"e1": ["a", "b"], "e2": ["a", "b", "c"], "e3": ["c"]})
     return H
+
+
+@pytest.fixture
+def hypergraph2():
+    H = xgi.Hypergraph()
+    H.add_nodes_from([0, "b", "c"])
```
Comment (leotrs) [inline tests/conftest.py:177]: I think the right test would be to add `["b", "c", 0]` i.e. the `0` last, so the (old) code would have overwritten `H._node[0] = H._node["b"]` and lost the neighbors of `0`. In this test, `0` is added first so it just gets relabeled to itself and there is no bug.

Comment (leotrs) [reply tests/conftest.py:177]: The new code looks better tho and I can't spot any bugs if any.

**Review by leotrs — APPROVED** [review]

**Review by leotrs — APPROVED** [review]

**Pull request merged**